src/core/system_impl.cpp

```diff
@@ -1381,7 +1381,7 @@ void SystemImpl::set_server_armed(bool armed)
 
 bool SystemImpl::is_server_armed() const
 {
-    return _parent.get_base_mode() & (MAV_MODE_FLAG_SAFETY_ARMED == MAV_MODE_FLAG_SAFETY_ARMED);
```

Comment:
The braces are wrong.

`(_parent.get_base_mode() & MAV_MODE_FLAG_SAFETY_ARMED) == MAV_MODE_FLAG_SAFETY_ARMED)`

Reply:
But look at my change, it's now anding and then `> 0`, so it should be true if the bit is there.

Reply:
Yes but it is a looser bitmask check. https://stackoverflow.com/a/14480058

Reply:
Ah, that's a thing, ok. Let me change it and force push.